src/jit/importer.cpp

```diff
@@ -6958,7 +6958,7 @@ bool  Compiler::impMethodInfo_hasRetBuffArg(CORINFO_METHOD_INFO * methInfo)
     //
   #ifdef FEATURE_HFA
     // On ARM HFAs are returned in registers.
-    if (!info.compIsVarArgs && IsHfa(methInfo->args.retTypeClass))
+    if (!info.compIsVarArgs && !opts.compUseSoftFP && IsHfa(methInfo->args.retTypeClass))
     {
```

Comment:
@briansull let me ask why that's not done for var args then?
Is the IsHfa just about arguments?